adapter/internal/oasparser/utils/url/url_utils.go

```diff
@@ -0,0 +1,12 @@
+package urlutils
+
+import "github.com/wso2/apk/adapter/internal/oasparser/constants"
+
+// GetURLType return URLType as https or http
+// depending on whether tls is enabled or not.
+func GetURLType(tlsEnabled bool) string {
+	if tlsEnabled {
```

Comment:
what if we add the protocol from the CRD directly? (HTTP/HTTPS/WS/WSS would be allowed values)

Reply:
Another possibility is to add a property in `Endpoint` struct called `tlsEnabled` and read it to enable TLS?